### PR TITLE
Lint project even if .ocamlformat is missing

### PR DESCRIPTION
**Pull request by CraigFe**
This resolves https://github.com/ocurrent/ocaml-ci/issues/107.

A simple solution that conditionally installs OCamlformat according to the result of the analysis phase.

### CODE REVIEW

File: lib/lint.ml [hunk 1]
```diff
@@ -2,52 +2,52 @@ open Current.Syntax
...
-        let img = ocamlformat_base_image ~base ~ocamlformat_source ~source in
+    |> (fun ocamlformat_source ->
+        let img = fmt_base_image ~base ~ocamlformat_source ~source in
         run_fmt ~img
       )
-    |> Current.map (function
-        | Some () -> `Checked
-        | None -> `Check_skipped
-      )
+    |> Current.map (fun () -> `Checked)
```
Comment (Julow) [inline lib/lint.ml]: Why not keeping `unit` here ?

File: lib/lint.ml [hunk 2]
```diff
@@ -2,52 +2,52 @@ open Current.Syntax
...
       let+ base = base and+ ocamlformat_source = ocamlformat_source in
-      ocamlformat_dockerfile ~base ~ocamlformat_source
+      fmt_dockerfile ~base ~ocamlformat_source
     in
-    Docker.build ~label:"OCamlformat" ~dockerfile source
+    Docker.build ~label:"fmt" ~dockerfile source
 
   let run_fmt ~img =
-    Docker.run ~label:"lint" img ~args:[ "sh"; "-c"; "dune build @fmt || (echo \"dune build @fmt failed\"; exit 2)" ]
+    Docker.run ~label:"lint-fmt" img ~args:[ "sh"; "-c"; "dune build @fmt || (echo \"dune build @fmt failed\"; exit 2)" ]
```
Comment (Julow) [inline lib/lint.ml]: What about a more descriptive label ? eg. "formatting"
This is shown to the users in the reports.

Comment (CraigFe) [reply lib/lint.ml]: I don't think this is shown to the users at any point. The user-facing name is given here: https://github.com/CraigFe/ocaml-ci/blob/lint-dune-files/service/pipeline.ml#L72

File: lib/lint.ml (hunk 1 above)
Comment (CraigFe) [reply lib/lint.ml]: `` `Checked`` is used by the `summarise` step (https://github.com/CraigFe/ocaml-ci/blob/lint-dune-files/service/pipeline.ml#L103) after constructing the list of jobs (https://github.com/CraigFe/ocaml-ci/blob/lint-dune-files/service/pipeline.ml#L53-L73).

File: lib/lint.ml (hunk 2 above)
Comment (Julow) [reply lib/lint.ml]: True

File: lib/lint.ml (hunk 1 above)
Comment (Julow) [reply lib/lint.ml]: I think the map `` unit -> `Checked `` should be moved there. It is used in a very specific logic that is internal to `summarise`. The fact that the variant have to be open in `Lint.v` signature shows that, I think.

**Comment by CraigFe** [comment]
Rebased over https://github.com/ocurrent/ocaml-ci/pull/108 to get the CI working.

**Comment by CraigFe** [comment]
Now rebased over https://github.com/ocurrent/ocaml-ci/pull/111 (merged) rather than https://github.com/ocurrent/ocaml-ci/pull/108.

**Comment by CraigFe** [comment]
Have locally verified that this works for `vocal-project/vocal#master`.

**Pull request merged**